NachoSpigot-Server/src/main/java/net/techcable/tacospigot/HopperPusher.java

```diff
@@ -8,29 +8,13 @@
 import net.minecraft.server.World;
 
 public interface HopperPusher {
-
+    
     public default TileEntityHopper findHopper() {
-        BlockPosition pos = new BlockPosition(getX(), getY(), getZ());
-        int startX = pos.getX() - 1;
-        int endX = pos.getX() + 1;
-        int startY = Math.max(0, pos.getY() - 1);
-        int endY = Math.min(255, pos.getY() + 1);
-        int startZ = pos.getZ() - 1;
-        int endZ = pos.getZ() + 1;
-        for (int x = startX; x <= endX; x++) {
-            for (int y = startY; y <= endY; y++) {
-                for (int z = startZ; z <= endZ; z++) {
-                    BlockPosition adjacentPos = new BlockPosition(x, y, z);
-                    TileEntityHopper hopper = HopperHelper.getHopper(getWorld(), adjacentPos);
-                    if (hopper == null) continue; // Avoid playing with the bounding boxes, if at all possible
-                    /**
-                     * We add one to getY(), just like {@link TileEntityHopper#b(IHopper)}
-                     */
-                    AxisAlignedBB hopperBoundingBox = hopper.getHopperLookupBoundingBox();
-                    if (hopperBoundingBox.b(this.getBoundingBox())) { // AxisAlignedBB.b(AxisAlignedBB) -> isIntersect()
-                        return hopper;
-                    }
-                }
+        int y1 = MathHelper.floor(getY());
+        for (int y = y1; y > y1 - 2; y--) {
+            TileEntityHopper hopper = HopperHelper.getHopper(getWorld(), new BlockPosition(getX(), y, getZ()));
```

Comment:
Would it help if we have getX() and getZ() as a local variable? Because now we're continuously calling getX() and getZ() which then calls another method to eventually get the variable address. Not sure if the java compiler fixes/inlines this though?

Reply:
maybe